kolibri/core/assets/src/core-actions.js

```diff
@@ -216,7 +216,7 @@ function getCurrentSession(store) {
   const id = 'current';
   const sessionModel = SessionResource.getModel(id);
   const sessionPromise = sessionModel.fetch({});
-  sessionPromise.then((session) => {
+  return sessionPromise.then((session) => {
```

Comment:
Not sure it'll work for your exact problem. But yes, here I returned a promise and then do the routing after the promise is resolved https://github.com/christianmemije/kolibri/blob/8191c3d09aeabdbb18b878a4ee224491db12ca3f/kolibri/plugins/user/assets/src/app.js#L15